//depot/google3/third_party/libffi/patches/401927869_compact_unwind_warning.patch#1 - opened for add
/google/src/cloud/dmaclach/libffi/google3/third_party/libffi/patches/401927869_compact_unwind_warning.patch - empty, assuming text.
Change 401927869 by dmaclach@dmaclach:libffi:18722:citc on 2021/10/08 22:42:27 *pending*

	Remove warning:
	
	ld: warning: could not create compact unwind for _ffi_call_unix64: does not use RBP or RSP based frame
	
	By following convention of saving the PC twice in CIE meaning disable compact unwind encoding. See 
	
	https://source.corp.google.com/piper///depot/google3/third_party/ld64/src/ld/parsers/libunwind/DwarfParser.hpp;rcl=321819767;l=778
	
	PRESUBMIT=passed
	FIXED=149760823
	R=ioanalexandru
	DELTA=20 (19 added, 0 deleted, 1 changed)
	DELTA_BY_EXTENSION=S=20
	MARKDOWN=true

Affected files ...

... //depot/google3/third_party/libffi/darwin_ios/src/x86/unix64_x86_64.S#1 edit
... //depot/google3/third_party/libffi/darwin_osx/src/x86/unix64_x86_64.S#1 edit
... //depot/google3/third_party/libffi/src/x86/unix64.S#2 edit

==== //depot/google3/third_party/libffi/darwin_ios/src/x86/unix64_x86_64.S#1 - /google/src/cloud/dmaclach/libffi/google3/third_party/libffi/darwin_ios/src/x86/unix64_x86_64.S ====
--- /google/src/files/206900402/depot/google3/third_party/libffi/darwin_ios/src/x86/unix64_x86_64.S	2018-08-01 05:25:06.000000000 -0400
+++ /google/src/cloud/dmaclach/libffi/google3/third_party/libffi/darwin_ios/src/x86/unix64_x86_64.S	2021-10-11 11:52:52.000000000 -0400
@@ -443,6 +443,13 @@
 	.byte	0x1b			/* FDE Encoding (pcrel sdata4) */
 	.byte	0xc, 7, 8		/* DW_CFA_def_cfa, %rsp offset 8 */
 	.byte	0x80+16, 1		/* DW_CFA_offset, %rip offset 1*-8 */
+#ifdef __APPLE__
+   /* ld64 has an idiom of PC saved twice in CIE meaning disable compact unwind
+      encoding. This prevents warning:
+      ld: warning: could not create compact unwind for _ffi_call_unix64: does not use RBP or RSP based frame
+      does not use RBP or RSP based frame". */
+	.byte	0x80+16, 1
+#endif
 	.balign 8
 L(ECIE):
 
==== //depot/google3/third_party/libffi/darwin_osx/src/x86/unix64_x86_64.S#1 - /google/src/cloud/dmaclach/libffi/google3/third_party/libffi/darwin_osx/src/x86/unix64_x86_64.S ====
--- /google/src/files/206900402/depot/google3/third_party/libffi/darwin_osx/src/x86/unix64_x86_64.S	2018-08-01 05:25:06.000000000 -0400
+++ /google/src/cloud/dmaclach/libffi/google3/third_party/libffi/darwin_osx/src/x86/unix64_x86_64.S	2021-10-11 11:53:05.000000000 -0400
@@ -443,6 +443,11 @@
 	.byte	0x1b			/* FDE Encoding (pcrel sdata4) */
 	.byte	0xc, 7, 8		/* DW_CFA_def_cfa, %rsp offset 8 */
 	.byte	0x80+16, 1		/* DW_CFA_offset, %rip offset 1*-8 */
+#ifdef __APPLE__
+   /* ld64 has an idiom of PC saved twice in CIE meaning disable compact unwind
+      encoding. This prevents a warning "does not use RBP or RSP based frame" */
+	.byte	0x80+16, 1
+#endif
 	.balign 8
 L(ECIE):
 
==== //depot/google3/third_party/libffi/src/x86/unix64.S#2 - /google/src/cloud/dmaclach/libffi/google3/third_party/libffi/src/x86/unix64.S ====
--- /google/src/files/206900402/depot/google3/third_party/libffi/src/x86/unix64.S	2018-08-01 05:25:06.000000000 -0400
+++ /google/src/cloud/dmaclach/libffi/google3/third_party/libffi/src/x86/unix64.S	2021-10-11 11:53:05.000000000 -0400
@@ -441,6 +441,13 @@
 	.byte	0x1b			/* FDE Encoding (pcrel sdata4) */
 	.byte	0xc, 7, 8		/* DW_CFA_def_cfa, %rsp offset 8 */
 	.byte	0x80+16, 1		/* DW_CFA_offset, %rip offset 1*-8 */
+#ifdef __APPLE__
+   /* ld64 has an idiom of PC saved twice in CIE meaning disable compact unwind
+      encoding. This prevents warning:
+      ld: warning: could not create compact unwind for _ffi_call_unix64: does not use RBP or RSP based frame
+   */
+	.byte	0x80+16, 1
+#endif
 	.balign 8
 L(ECIE):
 
